build.ps1

```diff
@@ -429,6 +429,10 @@ function Build-CMakeProject
     $SwiftArgs.Add("-Xlinker /INCREMENTAL:NO") | Out-Null
     $SwiftArgs.Add("-Xlinker /DEBUG:DWARF") | Out-Null
 
+    # Turn off safeseh for lld as it has safeseh enabled by default
+    # and fails with an ICU data object file icudt69l_dat.obj.
+    $SwiftArgs.Add("-Xlinker /SAFESEH:NO") | Out-Null
```

Comment:
I don't think that we should disable Safe SEH globally, it has some benefits for attack mitigation.

Reply:
If it was disabled before the lld switch, not much was lost? Is there a way to disable it selectively?

Reply:
Maybe I misunderstood "globally". We could disable safe seh in the `Build-Foundation` case only.

Reply:
Yeah, I think that we should be able to do that.  We can add `-D CMAKE_SHARED_LINKER_FLAGS="/SAFESEH:NO"` to `Build-Foundation`.

Reply:
I saw that `CMAKE_SHARED_LINKER_FLAGS` had no effect on the swift linking. Updated it so it passes the option through a parameter instead.